staff/manuel-villalvilla/arrays/flat.js

```diff
@@ -0,0 +1,51 @@
+function flat (array, depth) {
+    if (Array.isArray(array)) { // compruebo si el array pasado es un array lo primero
+        if (depth && typeof depth === 'number') {
+            let newArr = flat(array); // funcion recursiva
+            for (let a = 1; a < depth; a++) {
+                newArr = flat(newArr);
+            }
+            return newArr;
+        } 
+        const newArray = [];
+        // let index = 0; 
+        for (let i = 0; i < array.length; i++) {
+            if (typeof array[i] === 'object') {
+                for (let j = 0; j < array[i].length; j++) {
+                    if (typeof array[i][j] === 'number') {
+                        newArray[newArray.length] = array[i][j];
+                        // index++;
+                    } else {
+                        newArray[newArray.length] = array[i][0];
+                        // index++;
+                    }
+                }
+            } else {
+                newArray[newArray.length] = array[i];
+                // index++;
+            }
+        }
+        return newArray;
+    }
+}
+
```

Comment:
Te dejo por acá algunos comentarios para que los conversemos después:
- Validación de parámetros.
- typeof vs instanceof en caso de arrays
- if else en el for anidado diferenciando 'numbers' del resto de los tipos de datos?